### ISSUE TITLE
Build problems

### ISSUE THREAD

**Issue by coyote036** [psieg/Lightpack#460]
I cant generate sln keep geting this errors.
I have try VS dev prompt from software folder, try editing build-config, update_local and other things that i found in Issus section.
Uve VS 2019 comunnity  and git clone. What do I do wrong?

C:\Users\Coyote01\source\repos\Lightpack\Software\scripts\win32>generate_sln.bat

C:\Users\Coyote01\source\repos\Lightpack\Software\scripts\win32>REM call this from $repo/Software

C:\Users\Coyote01\source\repos\Lightpack\Software\scripts\win32>\bin\qmake -recursive -tp vc Lightpack.pro
The system cannot find the path specified.

C:\Users\Coyote01\source\repos\Lightpack\Software\scripts\win32>REM qmake only allows shared linkage of the msvcr* runtime (for reasons acceptable when using Qt in the projects)

C:\Users\Coyote01\source\repos\Lightpack\Software\scripts\win32>REM for our lightweiht, non-qt components which we want to inject into other processes this is bad

C:\Users\Coyote01\source\repos\Lightpack\Software\scripts\win32>REM so after qmake is done, run this little script to rewrite the vcxproj's to use static linking

C:\Users\Coyote01\source\repos\Lightpack\Software\scripts\win32>powershell "ForEach($f in Get-ChildItem hooks\*.vcxproj) { (Get-Content $f).replace(\"MultiThreadedDLL\",\"MultiThreaded\") | Set-Content $f }"
Get-ChildItem : Cannot find path 'C:\Users\Coyote01\source\repos\Lightpack\Software\scripts\win32\hooks' because it
does not exist.
At line:1 char:15
+ ForEach($f in Get-ChildItem hooks\*.vcxproj) { (Get-Content $f).repla ...
+               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (C:\Users\Coyote...pts\win32\hooks:String) [Get-ChildItem], ItemNotFound
   Exception
    + FullyQualifiedErrorId : PathNotFound,Microsoft.PowerShell.Commands.GetChildItemCommand

C:\Users\Coyote01\source\repos\Lightpack\Software\scripts\win32>powershell "ForEach($f in Get-ChildItem unhook\*.vcxproj) { (Get-Content $f).replace(\"MultiThreadedDLL\",\"MultiThreaded\") | Set-Content $f }"
Get-ChildItem : Cannot find path 'C:\Users\Coyote01\source\repos\Lightpack\Software\scripts\win32\unhook' because it
does not exist.
At line:1 char:15
+ ForEach($f in Get-ChildItem unhook\*.vcxproj) { (Get-Content $f).repl ...
+               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (C:\Users\Coyote...ts\win32\unhook:String) [Get-ChildItem], ItemNotFound
   Exception
    + FullyQualifiedErrorId : PathNotFound,Microsoft.PowerShell.Commands.GetChildItemCommand

C:\Users\Coyote01\source\repos\Lightpack\Software\scripts\win32>powershell "ForEach($f in Get-ChildItem offsetfinder\*.vcxproj) { (Get-Content $f).replace(\"MultiThreadedDLL\",\"MultiThreaded\") | Set-Content $f }"
Get-ChildItem : Cannot find path 'C:\Users\Coyote01\source\repos\Lightpack\Software\scripts\win32\offsetfinder'
because it does not exist.
At line:1 char:15
+ ForEach($f in Get-ChildItem offsetfinder\*.vcxproj) { (Get-Content $f ...
+               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (C:\Users\Coyote...32\offsetfinder:String) [Get-ChildItem], ItemNotFound
   Exception
    + FullyQualifiedErrorId : PathNotFound,Microsoft.PowerShell.Commands.GetChildItemCommand

C:\Users\Coyote01\source\repos\Lightpack\Software\scripts\win32>REM qmake can't create Win32 projects in a x64 SUBDIRS sln. The files are Win32, but the SLN generator assumes all vcxproj's are x64 sln

C:\Users\Coyote01\source\repos\Lightpack\Software\scripts\win32>REM so we have to manually rewrite the SLN to reference the Win32 builds

C:\Users\Coyote01\source\repos\Lightpack\Software\scripts\win32>powershell "$f = \"Lightpack.sln\"; $sln = Get-Content $f; $guid = ($sln | select-string -Pattern 'prismatik-hooks32.*({[0-9A-Z\-]*}).*').Matches[0].Groups[1].Value; $sln = $sln -replace(\"($guid.*)x64\",'$1Win32'); $guid = ($sln | select-string -Pattern 'prismatik-unhook32.*({[0-9A-Z\-]*}).*').Matches[0].Groups[1].Value; $sln = $sln -replace(\"($guid.*)x64\",'$1Win32'); $guid = ($sln | select-string -Pattern 'offsetfinder.*({[0-9A-Z\-]*}).*').Matches[0].Groups[1].Value; $sln = $sln -replace(\"($guid.*)x64\",'$1Win32'); $sln | Set-Content $f"
Get-Content : Cannot find path 'C:\Users\Coyote01\source\repos\Lightpack\Software\scripts\win32\Lightpack.sln' because
it does not exist.
At line:1 char:30
+ $f = "Lightpack.sln"; $sln = Get-Content $f; $guid = ($sln | select-s ...
+                              ~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (C:\Users\Coyote...2\Lightpack.sln:String) [Get-Content], ItemNotFoundEx
   ception
    + FullyQualifiedErrorId : PathNotFound,Microsoft.PowerShell.Commands.GetContentCommand

Cannot index into a null array.
At line:1 char:46
+ ... Content $f; $guid = ($sln | select-string -Pattern 'prismatik-hooks32 ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : NullArray

Cannot index into a null array.
At line:1 char:201
+ ... '$1Win32'); $guid = ($sln | select-string -Pattern 'prismatik-unhook3 ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : NullArray

Cannot index into a null array.
At line:1 char:357
+ ... '$1Win32'); $guid = ($sln | select-string -Pattern 'offsetfinder.*({[ ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : NullArray

**Comment by zomfg** [comment]
> C:\Users\Coyote01\source\repos\Lightpack\Software\scripts\win32>generate_sln.bat

should be
```
C:\Users\Coyote01\source\repos\Lightpack\Software>.\scripts\win32\generate_sln.bat
```

**Comment by coyote036** [comment]
C:\Users\Coyote01\source\repos\Lightpack\Software\scripts\win32>C:\Users\Coyote01\source\repos\Lightpack\Software>.\scripts\win32\generate_sln.bat
The system cannot find the path specified.

**Comment by coyote036** [comment]
C:\Users\Coyote01\source\repos\Lightpack\Software>powershell "$f = \"Lightpack.sln\"; $sln = Get-Content $f; $guid = ($sln | select-string -Pattern 'prismatik-hooks32.*({[0-9A-Z\-]*}).*').Matches[0].Groups[1].Value; $sln = $sln -replace(\"($guid.*)x64\",'$1Win32'); $guid = ($sln | select-string -Pattern 'prismatik-unhook32.*({[0-9A-Z\-]*}).*').Matches[0].Groups[1].Value; $sln = $sln -replace(\"($guid.*)x64\",'$1Win32'); $guid = ($sln | select-string -Pattern 'offsetfinder.*({[0-9A-Z\-]*}).*').Matches[0].Groups[1].Value; $sln = $sln -replace(\"($guid.*)x64\",'$1Win32'); $sln | Set-Content $f"
Get-Content : Cannot find path 'C:\Users\Coyote01\source\repos\Lightpack\Software\Lightpack.sln' because it does not
exist.
At line:1 char:30
+ $f = "Lightpack.sln"; $sln = Get-Content $f; $guid = ($sln | select-s ...
+                              ~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (C:\Users\Coyote...e\Lightpack.sln:String) [Get-Content], ItemNotFoundEx
   ception
    + FullyQualifiedErrorId : PathNotFound,Microsoft.PowerShell.Commands.GetContentCommand

Cannot index into a null array.
At line:1 char:46
+ ... Content $f; $guid = ($sln | select-string -Pattern 'prismatik-hooks32 ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : NullArray

Cannot index into a null array.
At line:1 char:201
+ ... '$1Win32'); $guid = ($sln | select-string -Pattern 'prismatik-unhook3 ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : NullArray

Cannot index into a null array.
At line:1 char:357
+ ... '$1Win32'); $guid = ($sln | select-string -Pattern 'offsetfinder.*({[ ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : NullArray

**Comment by zomfg** [comment]
let me rephrase
`cd` to `Software` directory
run `.\scripts\win32\generate_sln.bat`

**Comment by coyote036** [comment]
C:\Users\Coyote01\source\repos\Lightpack\Software>.\scripts\win32\generate_sln.bat

C:\Users\Coyote01\source\repos\Lightpack\Software>REM call this from $repo/Software

C:\Users\Coyote01\source\repos\Lightpack\Software>\bin\qmake -recursive -tp vc Lightpack.pro
The system cannot find the path specified.

C:\Users\Coyote01\source\repos\Lightpack\Software>REM qmake only allows shared linkage of the msvcr* runtime (for reasons acceptable when using Qt in the projects)

C:\Users\Coyote01\source\repos\Lightpack\Software>REM for our lightweiht, non-qt components which we want to inject into other processes this is bad

C:\Users\Coyote01\source\repos\Lightpack\Software>REM so after qmake is done, run this little script to rewrite the vcxproj's to use static linking

C:\Users\Coyote01\source\repos\Lightpack\Software>powershell "ForEach($f in Get-ChildItem hooks\*.vcxproj) { (Get-Content $f).replace(\"MultiThreadedDLL\",\"MultiThreaded\") | Set-Content $f }"

C:\Users\Coyote01\source\repos\Lightpack\Software>powershell "ForEach($f in Get-ChildItem unhook\*.vcxproj) { (Get-Content $f).replace(\"MultiThreadedDLL\",\"MultiThreaded\") | Set-Content $f }"

C:\Users\Coyote01\source\repos\Lightpack\Software>powershell "ForEach($f in Get-ChildItem offsetfinder\*.vcxproj) { (Get-Content $f).replace(\"MultiThreadedDLL\",\"MultiThreaded\") | Set-Content $f }"

C:\Users\Coyote01\source\repos\Lightpack\Software>REM qmake can't create Win32 projects in a x64 SUBDIRS sln. The files are Win32, but the SLN generator assumes all vcxproj's are x64 sln

C:\Users\Coyote01\source\repos\Lightpack\Software>REM so we have to manually rewrite the SLN to reference the Win32 builds

C:\Users\Coyote01\source\repos\Lightpack\Software>powershell "$f = \"Lightpack.sln\"; $sln = Get-Content $f; $guid = ($sln | select-string -Pattern 'prismatik-hooks32.*({[0-9A-Z\-]*}).*').Matches[0].Groups[1].Value; $sln = $sln -replace(\"($guid.*)x64\",'$1Win32'); $guid = ($sln | select-string -Pattern 'prismatik-unhook32.*({[0-9A-Z\-]*}).*').Matches[0].Groups[1].Value; $sln = $sln -replace(\"($guid.*)x64\",'$1Win32'); $guid = ($sln | select-string -Pattern 'offsetfinder.*({[0-9A-Z\-]*}).*').Matches[0].Groups[1].Value; $sln = $sln -replace(\"($guid.*)x64\",'$1Win32'); $sln | Set-Content $f"
Get-Content : Cannot find path 'C:\Users\Coyote01\source\repos\Lightpack\Software\Lightpack.sln' because it does not
exist.
At line:1 char:30
+ $f = "Lightpack.sln"; $sln = Get-Content $f; $guid = ($sln | select-s ...
+                              ~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (C:\Users\Coyote...e\Lightpack.sln:String) [Get-Content], ItemNotFoundEx
   ception
    + FullyQualifiedErrorId : PathNotFound,Microsoft.PowerShell.Commands.GetContentCommand

Cannot index into a null array.
At line:1 char:46
+ ... Content $f; $guid = ($sln | select-string -Pattern 'prismatik-hooks32 ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : NullArray

Cannot index into a null array.
At line:1 char:201
+ ... '$1Win32'); $guid = ($sln | select-string -Pattern 'prismatik-unhook3 ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : NullArray

Cannot index into a null array.
At line:1 char:357
+ ... '$1Win32'); $guid = ($sln | select-string -Pattern 'offsetfinder.*({[ ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : NullArray

**Comment by zomfg** [comment]
you skipped the very first prerequisite: `QTDIR`
https://github.com/psieg/Lightpack#prismatik-build-instructions-for-windows

**Comment by coyote036** [comment]
I have installed QT 6.0.3 open source edition and in my QT dir there is not msvc_xxxx folder only mingw81_64 folder.

**Comment by zomfg** [comment]
Qt6 wont work until 6.2 I think, get 5.15 for now
![image](https://user-images.githubusercontent.com/239811/115597843-05465e00-a2da-11eb-910f-a43e658dea8b.png)

**Comment by coyote036** [comment]
Ok, thx i will try to get 5.15 and post resolts. This is now the same problem like https://github.com/psieg/Lightpack/issues/391#

**Comment by coyote036** [comment]
Can't find Qt 5.15 installer only src I download 5.12 and there is not msvc folder.
 Is there any other way or different development tool do use ?

**Comment by zomfg** [comment]
https://www.qt.io/download-qt-installer

**Comment by coyote036** [comment]
I have download QT 5.14.2 select MSVC2017 64bit, in Environment Variables added %QTDIR% with out %, edit build-vars.prf to only win32 {
# Set this to x86 to compile 32-bit
	TARGET_ARCH = x86_64
}
and from Developer command promp run from software folder .\scripts\win32\generate_sln.bat and manage to get .sln file without errors.
I did not try it in VS, will post resolts later.

**Comment by coyote036** [comment]
The agony continues :)
QT 5.14.2 use MSVC2017 so noq i get this errors in VS when run debugger, it seems that I must use QT 5.15 becouse VS2019.

Severity	Code	Description	Project	File	Line	Suppression State
Error	MSB3073	The command "cd C:\Users\Coyote01\source\repos\Lightpack\Software\bin\ 
if errorlevel 1 goto VCEnd
C:/Qt/Qt5.14.2/5.14.2/msvc2017_64/bin/windeployqt --no-angle --no-svg --no-translations --no-compiler-runtime "Prismatik.exe" 
if errorlevel 1 goto VCEnd
if 142 LEQ 140 copy /y "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\redist\x64\Microsoft.VC142.CRT\msvcp142.dll" .\ 
if errorlevel 1 goto VCEnd
if 142 GTR 140 copy /y "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Redist\MSVC\14.28.29910\x64\Microsoft.VC142.CRT\msvcp*.dll" .\ 
if errorlevel 1 goto VCEnd
if 142 LSS 140 copy /y "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\redist\x64\Microsoft.VC142.CRT\msvcr142.dll" .\ 
if errorlevel 1 goto VCEnd
if 142 EQU 140 copy /y "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\redist\x64\Microsoft.VC142.CRT\vcruntime142.dll" .\ 
if errorlevel 1 goto VCEnd
if 142 GTR 140 copy /y "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Redist\MSVC\14.28.29910\x64\Microsoft.VC142.CRT\vcruntime*.dll" .\ 
if errorlevel 1 goto VCEnd
copy /y "\libcrypto-1_1-x64.dll" .\ 
if errorlevel 1 goto VCEnd
copy /y "\libssl-1_1-x64.dll" .\ 
if errorlevel 1 goto VCEnd
IF EXIST "\msvcr*.dll" copy /y "\msvcr*.dll" .\ 
if errorlevel 1 goto VCEnd

:VCEnd" exited with code 1.	Prismatik	C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets	155

**Comment by zomfg** [comment]
you can disable ssl requirement https://github.com/psieg/Lightpack/issues/458#issuecomment-820950550

**Comment by coyote036** [comment]
Yes I disable ssl, in my build-vars.prf there is only TARGET_ARCH = x86_64 and i re-run generate_sln

**Comment by zomfg** [comment]
look at my example in the link

**Comment by coyote036** [comment]
It is working, thanks for your dedicated time to help me.

**Comment by psieg** [comment]
Thanks @zomfg for helping
